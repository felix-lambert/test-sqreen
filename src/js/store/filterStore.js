/**
 * This is a reducer, a pure function with (state, action) => state signature.
 * It describes how an action transforms the state into the next state.
 * you write a special function called a reducer to decide how every 
 * action transforms the entire application's state
 */
export const filter = (state = '', action) => {
  if (action) {
    switch (action.type) {
      case 'VISIBILITY_FILTER':
        return action.text;
      default:
        return state;
    }
  }
  return '';
};