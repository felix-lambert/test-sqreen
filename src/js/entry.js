/* React well understood the fact that the DOM is slow, and so needs to limit the interaction
   with it. It uses a virtual DOM. A Virtual DOM is a representation of the DOM in JavaScript. 
   Instead of generating the DOM itself as with a templating language, ie instead of dialoging 
   with the browser APIs to build the DOM, we only generate a tree of JavaScript objects in 
   memory.
*/
import React from 'react'

/*
  The react-dom package provides DOM-specific methods that can 
  be used at the top level of your app and as an escape hatch 
  to get outside of the React model if you need to
 */
import ReactDOM from 'react-dom'

import { store } from './store/'

/* 
  You may wonder why we split up the files at all — why 
  not keep everything in index.js? The reason is because 
  we need to import the components into our test, and if 
  we import them from the index.js file, ReactDOM.render() 
  will execute. This causes us to be dependent on the 
  existence of a DOM, even though most of our tests 
  won't need it (because they're using shallow rendering).
 */
import FilterableUserTable from './component/FilterableUserTableComponent';

export const USERS = [
  {id:1, name: 'Jennifer Tanker', email: 'qa_jennifer@tanker.io'},
  {id:2, name: 'Micher Tanker', email: 'qa_michel@tanker.io'},
  {id:3, name: 'Germaine Tanker', email: 'qa_germaine@tanker.io'},
  {id:4, 'name': 'Sebastien Tanker', 'email': 'qa_sebastien@tanker.io'},
  {id:5, 'name': 'QuentinAdmin', 'email': 'admin_quentin@tanker.io'},
  {id:6, 'name': 'DimitriAdmin Merejkowsky', 'email': 'admin_dimitri@tanker.io'},
  {id:7, 'name': '74bl€ Flp๔๕๛€Çç', 'email': 'demo_alois@tanker.io'},
  {id:8, 'name': 'DimitriAdmin Merejkowsky', 'email': 'admin_dimitri@tanker.io'},
  {id:9, 'name': 'TheoAdmin Delrieu', 'email': 'admin_theo@tanker.io'}
];

const render = () => {
  /* 
    With our knowledge so far, the only way to update the UI 
    is to create a new element, and pass it to ReactDOM.render()
    In practice, most React apps only call ReactDOM.render() once
   */
  ReactDOM.render(
    <FilterableUserTable
      /* 
        It retrieves the current state of the Redux store. 
        If we were on this, we're going to see '' because this 
        is the initial state of our application.
        The render function is called any time this store 
        state changes, so I can safely pass the current state 
        of this store as a prop to my root component.
       */
      filter={store.getState()}
      users={USERS}
    />,
    /*
      It renders it into the div I created inside the HTML. 
      It's div with the id called root
     */
    document.getElementById('root')
  );
};

/*
  You can use subscribe() to update the UI in response to state changes.
 */
store.subscribe(render);
render();