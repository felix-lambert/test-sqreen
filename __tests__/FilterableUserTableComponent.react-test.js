// Link.react-test.js
import React from 'react';
jest.unmock('../src/sass/styles.scss');
import FilterableUserTable from '../src/js/component/FilterableUserTableComponent';
import renderer from 'react-test-renderer';
import { store } from '../src/js/store/'

import USERS from '../src/json/users.json';

test('Filter gets state', () => {
  const component = renderer.create(
    <FilterableUserTable
      filter={store.getState()}
      users={USERS}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});