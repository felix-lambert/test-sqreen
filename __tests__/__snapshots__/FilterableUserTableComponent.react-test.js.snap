exports[`test Filter gets state 1`] = `
<div>
  <br />
  <div
    className="container">
    <table
      className="table table-striped">
      <thead>
        <tr>
          <th>
            MEMBERS IN MY TEAM
          </th>
          <th
            className="text-center">
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width="1em">
              <g>
                <path
                  d="m27.2 18.6q0-4.2-2.9-7.1t-7.1-2.9-7 2.9-3 7.1 2.9 7 7.1 3 7.1-3 2.9-7z m11.4 18.5q0 1.2-0.8 2.1t-2 0.8q-1.2 0-2-0.8l-7.7-7.7q-4 2.8-8.9 2.8-3.2 0-6.1-1.3t-5-3.3-3.4-5-1.2-6.1 1.2-6.1 3.4-5.1 5-3.3 6.1-1.2 6.1 1.2 5 3.3 3.4 5.1 1.2 6.1q0 4.9-2.7 8.9l7.6 7.6q0.8 0.9 0.8 2z" />
              </g>
            </svg>
            <input
              className="search"
              onChange={[Function]}
              placeholder="Search..." />
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  Jennifer Tanker
                  <br />
                  <small
                    className="logo">
                    qa_jennifer@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  Micher Tanker
                  <br />
                  <small
                    className="logo">
                    qa_michel@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  Germaine Tanker
                  <br />
                  <small
                    className="logo">
                    qa_germaine@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  Sebastien Tanker
                  <br />
                  <small
                    className="logo">
                    qa_sebastien@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  QuentinAdmin
                  <br />
                  <small
                    className="logo">
                    admin_quentin@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  DimitriAdmin Merejkowsky
                  <br />
                  <small
                    className="logo">
                    admin_dimitri@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  74bl€ Flp๔๕๛€Çç
                  <br />
                  <small
                    className="logo">
                    demo_alois@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  DimitriAdmin Merejkowsky
                  <br />
                  <small
                    className="logo">
                    admin_dimitri@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="row">
              <div
                className="col-md-2">
                <div
                  className="testimonial-desc">
                  <h1>
                    <img
                      alt=""
                      src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100" />
                  </h1>
                </div>
              </div>
              <div
                className="col-md-6 well-blue-p">
                <p>
                  TheoAdmin Delrieu
                  <br />
                  <small
                    className="logo">
                    admin_theo@tanker.io
                  </small>
                </p>
              </div>
            </div>
          </td>
          <td
            className="text-center logo">
            <h1>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m15 23.3c2.3 0 4.4-0.9 5.9-2.4 1.5-1.5 2.4-3.6 2.4-5.9s-0.9-4.4-2.4-5.9c-1.5-1.5-3.6-2.4-5.9-2.4s-4.4 0.9-5.9 2.4c-1.5 1.5-2.4 3.6-2.4 5.9s0.9 4.4 2.4 5.9c1.5 1.5 3.6 2.4 5.9 2.4z m0 11.7c5.9 0 10-1.7 10-3.3 0-3.4-3.9-6.7-10-6.7-6.2 0-10 3.3-10 6.7 0 1.6 3.8 3.3 10 3.3z m20-15h-3.3v-3.3c0-1-0.8-1.7-1.7-1.7s-1.7 0.7-1.7 1.7v3.3h-3.3c-0.9 0-1.7 0.7-1.7 1.7s0.8 1.6 1.7 1.6h3.3v3.4c0 0.9 0.8 1.6 1.7 1.6s1.7-0.7 1.7-1.6v-3.4h3.3c0.9 0 1.7-0.7 1.7-1.6s-0.8-1.7-1.7-1.7z" />
                </g>
              </svg>
            </h1>
            Click here
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
